src/pg_to_evalscript/javascript_common/common.js

```diff
@@ -44,3 +44,92 @@ function parse_rfc3339(dt, default_h = 0, default_m = 0, default_s = 0) {
 
   return result;
 }
+
+class ValidationError extends Error {
+  constructor(code, message) {
+    super(message);
+    this.code = code;
```

Comment:
I would change 

- `common.js`:
  - line 51: 
     - old: `this.code = code;`
     - new: `this.name = code;` 
   - rename `code` parameter to `name` in the whole `ValidationError`
- `utils.py`:
  - line 83: 
    - old: `expected = f"code: '{error_code}'" if error_code else error_message`
    - new: `expected = f"{error_code}:" if error_code else error_message` 
    - (so that it checks for the `<error_name>:` in ``<error_name>: <error_message>`)
- maybe other values would then be more appropriate for `VALIDATION_ERROR_CODES` codes

Reply:
Actually I extended Error and added custom property `code` because I didn't want to use name ( name is displayed when error is thrown), but I guess it doesn't  make much difference so I changed it as suggested.